File: CPE/Exam/231017/23F1_UVA11576.py
T = int(input())

for _ in range(1, T+1):
    n, m = map(int, input().split())
    words = [input() for _ in range(m)]

    ans = n * m
    for i in range(m-1):
        wa, wb = words[i], words[i+1]
        for i in range(n):
            if wa[i:] == wb[:n-i]:
                ans -= n - i
                break
    print(ans)

File: CPE/Exam/231017/23F1_UVA11576.cpp
#include <bits/stdc++.h>
using namespace std;
#define endl '\n'

int main() {
    ios::sync_with_stdio(false); cin.tie(nullptr); cout.tie(nullptr);
	int t, n, m, ans;
	cin >> t;
	while (t--) {
        ans = 0;
		cin >> n >> m;
		vector<string> words(m);
		for (int i=0; i<m; i++){
			cin >> words[i];
		}
		ans = n * m;
		for (int i=0; i<m-1; i++){
			for (int j=0; j<n; j++){
                if (words[i].substr(j) == words[i+1].substr(0, n-j)){
                    ans -= n - j;
                    break;
                }
            }
		}
        cout << ans << endl;
	}
}